chiron/neighbors.py

```diff
@@ -14,9 +14,31 @@
 
 
 class Space(ABC):
-    def __init__(self, box_vectors: Optional[jnp.array] = None) -> None:
+    def __init__(
+        self, box_vectors: Union[jnp.array, unit.Quantity, None] = None
+    ) -> None:
+        """
+        Abstract base class for defining the simulation space.
+
+        Parameters
+        ----------
+        box_vectors: jnp.array, optional
+            Box vectors for the system.
+        """
         if box_vectors is not None:
-            self.box_vectors = box_vectors
+            if isinstance(box_vectors, unit.Quantity):
+                if not box_vectors.unit.is_compatible(unit.nanometer):
+                    raise ValueError(
+                        f"Box vectors require distance unit, not {box_vectors.unit}"
+                    )
+                self.box_vectors = box_vectors.value_in_unit_system(unit.md_unit_system)
+            elif isinstance(box_vectors, jnp.ndarray):
+                if box_vectors.shape != (3, 3):
+                    raise ValueError(
+                        f"box_vectors should be a 3x3 array, shape provided: {box_vectors.shape}"
+                    )
+
+                self.box_vectors = box_vectors
 
```

Comment:
```suggestion
            else:
                raise ValueError(f'box_vectors need to be either `unit.Quantity` or `jnp.ndarray`')

```

Reply:
The code lacks a default case in the `if ... elif ...` statement, which could lead to unhandled situations. Added a suggestion for such a case.

Reply:
Good catch!